test-case/check-audio-equalizer.sh

```diff
@@ -44,103 +44,100 @@ func_test_eq()
-    if [ $is_iir -ne $is_fir ]; then
-        dlogi "no FIR or IIF filter is available in this pipeline, skip"
-        continue
-    fi
-
-    dlogi "0. Get amixer control id for IIR and FIR"
-    IIRid=`amixer -D hw:$sofcard controls | grep EQIIR| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
-    if [ -z $IIRid ]; then
-            dloge "can't find IIR filter"
-            exit 1
-    fi
-
-    FIRid=`amixer -D hw:$sofcard controls | grep EQFIR| head -1| sed 's/numid=\([0-9]*\),.*/\1/'`
-    if [ -z $FIRid ]; then
-            dloge "can't find FIR filter"
-            exit 1
+# this function performs IIR/FIR filter test
+# param1 must be must be iir or fir
+func_test_filter()
+{
+    local testfilter=$1
+    dlogi "1. Get amixer control id for $testfilter"
+    # TODO: Need to match alsa control id with the filter in the pipeline
+    Filterid=$(amixer -D hw:"$sofcard" controls | sed -n -e "/eq${testfilter}/I "'s/numid=\([0-9]*\),.*/\1/p' | head -1)
+    if [ -z "$Filterid" ]; then
+        dloge "can't find $testfilter"
+        exit 1
     fi
 
-    declare -a IIRList=($(ls -d ${current_dir}/eqctl/eq_iir_*.txt))
-    nIIRList=${#IIRList[@]}
-    dlogi "IIR list $nIIRList ${IIRList[*]}"
-    declare -a FIRList=($(ls -d ${current_dir}/eqctl/eq_fir_*.txt))
-    nFIRList=${#FIRList[@]}
-    dlogi "FIR list $nFIRList ${FIRList[*]}"
-    if [ $nIIRList ==  0 ] || [ $nFIRList ==  0 ]; then
-        dloge "IIR or FIR flter coeff list error!"
-	exit 1
+    declare -a FilterList=($(ls -d "${current_dir}"/eqctl/eq_"${testfilter}"_*.txt))
+    nFilterList=${#FilterList[@]}
+    dlogi "2. $testfilter list $nFilterList ${FilterList[*]}"
+    if [ "$nFilterList" -eq  0 ]; then
+        dloge "$testfilter flter coeff list error!"
+        exit 1
     fi
 
     for i in $(seq 1 $loop_cnt)
     do
-        dlogi "[$i/$loop_cnt] 1. Test IIR config list, IIR amixer control id=$IIRid"
-        for config in ${IIRList[@]}; do
-            func_test_eq $IIRid $current_dir/$config
-            result=$?
-            if [[ $result -ne 0 ]]; then
+        dlogi "[$i/$loop_cnt] 3. Test $testfilter config list, $testfilter amixer control id=$Filterid"
+        for config in "${FilterList[@]}"; do
+            func_test_eq "$Filterid" "$current_dir/$config" || {
                 dloge "Failed at $config"
-                let failed_cnt++
-            fi
+                : $((failed_cnt++))
+            }
         done
 
-        dlogi "[$i/$loop_cnt] 2. Test FIR config list, FIR amixer control id=$FIRid"
-        for config in ${FIRList[@]}; do
-            func_test_eq $FIRid $current_dir/$config
-            result=$?
-            if [[ $result -ne 0 ]]; then
-                dloge "Failed at $config"
-                let failed_cnt++
-            fi
-        done
-        dlogi "EQ test done: failed_cnt is $failed_cnt"
+        dlogi "$testfilter test done: failed_cnt is $failed_cnt"
         if [ $failed_cnt -gt 0 ]; then
             exit 1
```

Comment:
all failed case is exit, and if it hit at the end of line, either pass or passthrough(no filter). that's why `result` was initialized with 1.

Reply:
> all failed case is exit [immediately]

Why not run later tests when one test fail? Is it by design or just because you don't want to add an unrelated change in this PR? Either way is fine, I'm just curious.

Reply:
That's right, failed_cnt was there before. By that logic, it might fail on some coeffs in one filter. I prefer exit immediately at any failure but it's not harm.